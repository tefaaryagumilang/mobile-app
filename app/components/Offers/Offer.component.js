import React from 'react';
import PropTypes from 'prop-types';
import {View, Text} from 'react-native';
import styles from './Offers.styles';
import Touchable from '../Touchable.component';
import Image from 'react-native-image-progress';
import Bar from 'react-native-progress/Bar';
import {theme} from '../../styles/core.styles';
import {language} from '../../config/language';
import moment from 'moment';
import {getDayName, currencyFormatter} from '../../utils/transformer.util.js';
import SimasIcon from '../../assets/fonts/SimasIcon';

const loadError = () => <Text>{language.ERROR_MESSAGE__IMAGE_LOAD}</Text>;

const Offer = ({label, onPress, style = '', imgUrl, type, imageName, onLayout, loyaltyProgramType, discountAmount, permanentPercentageDiscount, startDate = '', endDate = '', validStartDate, validEndDate, iconPath, categoryLifeStyle, titleLifestyle, subTitleLifestyle, toogleTouchableLifestyle}) => {
  let offerDate = '';
  if (validStartDate || validEndDate) {
    offerDate = validStartDate === validEndDate ? language.DAY__EVERY + getDayName(validStartDate) :  `${moment(validStartDate, 'DD-MM-YYYY').format('DD MMM YYYY')}${' - '}${moment(validEndDate, 'DD-MM-YYYY').format('DD MMM YYYY')}`;
  }
  let showDate = '';
  if (startDate && endDate) {
    showDate = startDate === endDate ? language.DAY__EVERY + getDayName(startDate) : `${moment(startDate).format('D MMM YYYY')}${' - '}${ moment(endDate).format('D MMM YYYY')}`;
  }
  let titleOffer = '';
  if (offerDate) {
    titleOffer = 'Simas Catalog - ' + label;
  } else {
    titleOffer = 'Simas Catalog - ' + titleLifestyle;
  }
  return (
    <Touchable dtActionName={titleOffer} style={[styles.offerContainer, style]} onPress={onPress}>
      {
        type === 'offer' ?
          <View>
            <Image source={(toogleTouchableLifestyle) && categoryLifeStyle ? {uri: iconPath} : {uri: imgUrl}} resizeMode={'stretch'} renderError={loadError} indicator={Bar} indicatorProps={{
              showsText: true,
              color: theme.brand,
              size: 50,
              thickness: 2
            }} style={styles.offerImage} />
            {offerDate ?
              <View style={styles.offerDetails}>
                <View style={styles.iconContainer}><SimasIcon name='time-black' size={30} style={styles.iconStyleBlack}/></View>
                <View>
                  <Text style={styles.label}>{language.OFFER__BANNER_VALID_DATE}</Text>
                  <Text style={styles.labelValidDate}>{offerDate}</Text>
                </View>
              </View>
              : null}
            {(toogleTouchableLifestyle) && categoryLifeStyle ?
              <View style={styles.offerDetails}>
                <View>
                  <Text style={styles.label}>{titleLifestyle}</Text>
                  <Text style={styles.labelValidDate}>{subTitleLifestyle}</Text>
                </View>
              </View>
              : null}
          </View>
          :
          <View>
            <View>
              <Image source={{uri: 'https://' + imageName}} resizeMode={'stretch'} renderError={loadError} indicator={Bar} indicatorProps={{
                showsText: true,
                color: theme.brand,
                size: 50,
                thickness: 2
              }} style={styles.offerImage} />
            </View>
            <View style={styles.rowContainer}>
              <View onLayout={onLayout} style={styles.promoAmountContainer}>
                <Text style={styles.discountType}>{loyaltyProgramType === 'POINTS' ? 'VOUCHER' : 'DISCOUNT'}</Text>
                <Text style={styles.discountAmount}>{loyaltyProgramType === 'POINTS' ? 'Rp ' +  currencyFormatter(discountAmount) : permanentPercentageDiscount + '%'}</Text>
              </View>
              <View style={styles.qrTag}>
                <Text style={styles.qrTagText}>{language.PAY_BY_QR__TAG}</Text>
              </View>
            </View>
            <View style={styles.promoDetailContainer}>
              <View style={styles.promoDetailDateContainer}>
                <Text style={styles.boldText}>{label}</Text>
                <Text style={styles.label}>{showDate}</Text>
              </View>
            </View>
          </View>
      }

    </Touchable>
  );
};

Offer.propTypes = {
  label: PropTypes.string,
  imgUrl: PropTypes.string,
  subtitle: PropTypes.string,
  onPress: PropTypes.func,
  featureIconName: PropTypes.string,
  footer: PropTypes.string,
  style: PropTypes.object,
  startDate: PropTypes.number,
  endDate: PropTypes.number,
  type: PropTypes.string,
  imageName: PropTypes.string,
  onLayout: PropTypes.func,
  loyaltyProgramType: PropTypes.string,
  discountAmount: PropTypes.number,
  permanentPercentageDiscount: PropTypes.number,
  validStartDate: PropTypes.string,
  validEndDate: PropTypes.string,
  iconPath: PropTypes.string,
  categoryLifeStyle: PropTypes.bool,
  titleLifestyle: PropTypes.string,
  subTitleLifestyle: PropTypes.string,
  toogleTouchableLifestyle: PropTypes.bool,
};

export default Offer;
