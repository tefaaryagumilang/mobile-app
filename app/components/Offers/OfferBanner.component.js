import React from 'react';
import PropTypes from 'prop-types';
import {View, Text, Image} from 'react-native';
import styles from './Offers.styles';
import Touchable from '../Touchable.component';
import Bar from 'react-native-progress/Bar';
import {theme} from '../../styles/core.styles';
import {language} from '../../config/language';
import noBanner from '../../assets/images/no-banner.jpg';
import SimasIcon from '../../assets/fonts/SimasIcon';

const loadError = () => <Text>{language.ERROR_MESSAGE__IMAGE_LOAD}</Text>;

const Offer = ({onPress, style = '', imgUrl2, type, goToOffer, openDrawer}) => (
  <View>
    <Touchable style={[styles.offerBannerContainer, style]} onPress={onPress}>
      {
        type === 'offer' ?
          <View>
            <Image source={{uri: imgUrl2}} resizeMode={'stretch'} renderError={loadError} indicator={Bar} indicatorProps={{
              showsText: true,
              color: theme.brand,
              size: 50,
              thickness: 2
            }} style={styles.offerBannerContainer} />
            <Touchable style={styles.seeAll} onPress={goToOffer}>
              <Text style={styles.seeAlltext}>{language.OFFER__BANNER_SEEALL}</Text>
            </Touchable>
            <Touchable style={styles.burgerMenuContainer} onPress={openDrawer}>
              <View>
                <SimasIcon name={'Burger'} size={20} style={styles.burger}/>
              </View>
            </Touchable>
          </View>
          :
          <View>
            <View>
              <Image source={noBanner} resizeMode={'stretch'} renderError={loadError} indicator={Bar} indicatorProps={{
                showsText: true,
                color: theme.brand,
                size: 50,
                thickness: 2
              }} style={styles.offerBannerContainer} />
              <Touchable style={styles.burgerMenuContainer} onPress={openDrawer}>
                <View>
                  <SimasIcon name={'Burger'} size={20} style={styles.burger}/>
                </View>
              </Touchable>
            </View>
          </View>
      }

    </Touchable>
    
  </View>
);

Offer.propTypes = {
  label: PropTypes.string,
  imgUrl2: PropTypes.string,
  subtitle: PropTypes.string,
  onPress: PropTypes.func,
  featureIconName: PropTypes.string,
  footer: PropTypes.string,
  style: PropTypes.object,
  startDate: PropTypes.number,
  endDate: PropTypes.number,
  type: PropTypes.string,
  imageName: PropTypes.string,
  onLayout: PropTypes.func,
  loyaltyProgramType: PropTypes.string,
  discountAmount: PropTypes.number,
  permanentPercentageDiscount: PropTypes.number,
  goToOffer: PropTypes.func,
  openDrawer: PropTypes.func,
};

export default Offer;
