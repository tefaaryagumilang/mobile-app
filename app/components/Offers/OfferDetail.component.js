import React from 'react';
import PropTypes from 'prop-types';
import {View, Text, ScrollView, Linking} from 'react-native';
import noop from 'lodash/noop';
import {language} from '../../config/language';
import Image from 'react-native-image-progress';
import Bar from 'react-native-progress/Bar';
import styles from './OfferDetail.styles';
import {SinarmasButton} from '../FormComponents';
import result from 'lodash/result';
import SimasIcon from '../../assets/fonts/SimasIcon';
import moment from 'moment';
import {Toast} from '../../utils/RNHelpers.util.js';
import AutoHeightWebView from 'react-native-autoheight-webview';
import includes from 'lodash/includes';
import {Platform} from 'react-native';
let adjustAndroid;

if (Platform.OS === 'android') {
  adjustAndroid = require('react-native-adjust');
}

const loadError = () => <Text>{language.ERROR_MESSAGE__IMAGE_LOAD}</Text>;

class tabDetail extends React.Component {
  static propTypes = {
    offer: PropTypes.object,
    onOfferClick: PropTypes.func,
    cif: PropTypes.string,
    isOfferNilaiQ: PropTypes.bool,
  }

  openLink = (data) => () => {
    const link = result(data, '0', 'https://www.banksinarmas.com/PersonalBanking/NoPage');
    Linking.canOpenURL(link).then((supported) => {
      if (supported) {
        Linking.openURL(link);
      } else {
        Toast.show(language.ERROR_MESSAGE__CANNOT_HANDLE_URL, Toast.LONG);
      }
    });
  };

  render () {
    const {offer = {}, onOfferClick = noop, cif, isOfferNilaiQ} = this.props;
    const navigationKeys = result(offer, 'navigateTo', []);
    const offerStartDate = moment(offer.validStartDate, 'DD-MM-YYYY').format('DD MMM YYYY');
    const offerEndDate = moment(offer.validEndDate, 'DD-MM-YYYY').format('DD MMM YYYY');
    const navigationUrl = result(offer, 'navigateTo.0', '');
    const url = navigationUrl.substring(0, 4);
    const productName = navigationUrl;
    const tncStyle = '<style>* body {font-family: roboto; font-size: 14;}</style>';
    const headingStyle = '<style>* body {font-family: roboto; font-size: 14; font-weight: bold;}</style>';
    const bannerDetail = includes(offer.label, ('<p>')) || includes(offer.label, ('<li>')) || includes(offer.label, ('<br>')) || includes(offer.label, ('<html>'));
    const label = result(offer, 'label', '');
    let adjustEvent;
    if (Platform.OS === 'android') {
      adjustEvent = new adjustAndroid.AdjustEvent('bdf0nz');
      adjustEvent.addCallbackParameter('page_id', 'ac-spe-1-1');
      adjustEvent.addCallbackParameter('cif', cif);
      adjustEvent.addCallbackParameter('page_name', label);
      adjustAndroid.Adjust.trackEvent(adjustEvent);
    }
    return (
      <View style={styles.container}>
        <ScrollView>
          <View style={styles.fullContainer}>
            <Image source={{uri: offer.imgUrl}} resizeMode={'stretch'} renderError={loadError} indicator={Bar} indicatorProps={styles.indicator} style={styles.offerImage} />
            <View style={styles.detailContainer}>
              { bannerDetail === true ?
                <View style={styles.infoContainer}>
                  {offer.label ? <View style={styles.labelContainerHTML}><AutoHeightWebView style={styles.webViewWidth} source={{html: headingStyle + offer.label.split('<li>')[0] || offer.label.split('<br>')[0]}}/></View> : null}
                </View>
                :
                <View style={styles.infoContainer}>
                  {offer.label ? <View style={styles.labelContainer}><Text style={styles.labelFont}>{offer.label.split('\n')[0]}</Text></View> : null}
                </View>
              }
              {offer.validStartDate ?
                <View style={styles.offerDetails}>
                  <View style={styles.iconContainer}><SimasIcon name='time-black' size={30} style={styles.iconStyleRed}/></View>
                  <View>
                    <Text style={styles.labelValidDate}>{language.OFFER__BANNER_VALID_DATE}</Text>
                    <Text style={styles.label}>{offerStartDate} - {offerEndDate}</Text>
                  </View>
                </View>
                : null }
            </View>
            <View style={styles.borderGreyTop}/>
            <View style={styles.detailContainer}>
              { bannerDetail === true ?
                <View style={styles.infoContainerTnC}>
                  {offer.label ? <View style={styles.labelContainerHTML}><AutoHeightWebView style={styles.webViewWidth} source={{html: tncStyle + offer.label}}/></View> : null}
                </View>
                :
                <View style={styles.infoContainerTnC}>
                  {offer.label ? <View style={styles.labelContainer}><Text style={styles.label}>{offer.label}</Text></View> : null}
                  {offer.subtitle === '' || offer.subtitle === '-' ? null : <Text style={styles.subtitle}>{offer.subtitle}</Text> }
                  {offer.footer === '' || offer.footer === '-' ? null : <Text style={styles.footer}>{offer.footer}</Text>}
                </View> }
            </View>
          </View>
        </ScrollView>
        { result(navigationKeys, '0', '') === 'NONE' || result(navigationKeys, '0', '') === '' ?
          null
          : url === 'http' || url === 'HTTP' ?
            <View style={styles.buttonContainer}>
              <SinarmasButton text={language.QR_PROMO__CHECK_THIS_OUT} onPress={this.openLink(navigationKeys)}/>
            </View>
            :
            isOfferNilaiQ ?
              <View style={styles.buttonContainer}>
                <SinarmasButton text={offer.textDescription} onPress={onOfferClick(productName)}/>
              </View>
              :
              <View style={styles.buttonContainer}>
                <SinarmasButton text={language.QR_PROMO__CHECK_THIS_OUT} onPress={onOfferClick(productName)}/>
              </View>
        }
      </View>
    );
  }
}

export default tabDetail;
