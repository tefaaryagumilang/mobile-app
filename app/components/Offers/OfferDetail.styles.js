import {theme} from '../../styles/core.styles';
import {Dimensions} from 'react-native';

const {width} = Dimensions.get('window');
const trueWidth = width;
const trueHeight = (trueWidth * 7) / 16;

export default {
  container: {
    backgroundColor: theme.white,
    flex: 1,
    justifyContent: 'space-between'
  },
  detailContainer: {
    backgroundColor: theme.white
  },
  infoContainer: {
    paddingHorizontal: 10,
  },
  offerImage: {
    width: trueWidth,
    height: trueHeight
  },
  labelContainer: {
    paddingTop: 20
  },
  indicator: {
    showsText: true,
    color: theme.brand,
    size: 50,
    thickness: 2
  },
  buttonContainer: {
    backgroundColor: theme.white,
    paddingVertical: 20,
    paddingHorizontal: 10
  },
  fullContainer: {
    paddingBottom: 20
  },
  greyLine: {
    borderTopWidth: 1,
    borderColor: theme.greyLine,
    paddingBottom: 5,
    paddingTop: 5,
    paddingHorizontal: 50
  },
  borderGreyTop: {
    backgroundColor: theme.greyLine,
    height: 10,
    paddingHorizontal: 50
  },
  offerDetails: {
    flexDirection: 'row',
    marginHorizontal: 20,
    paddingBottom: 20
  },
  label: {
    marginHorizontal: 10,
    color: theme.black
  },
  labelValidDate: {
    marginHorizontal: 10,
    marginTop: 10,
    color: theme.black,
  },
  labelFont: {
    marginHorizontal: 10,
    fontWeight: 'bold',
    fontSize: theme.fontSizeMediumStyle,
    color: theme.black,
    fontFamily: 'Roboto'
  },
  iconContainer:
  {
    marginTop: 10,
  },
  infoContainerTnC: {
    paddingHorizontal: 10,
  },
  webViewWidth: {
    width: width - 40,
  },
  labelContainerHTML: {
    paddingTop: 20,
    paddingHorizontal: 10
  }
};
