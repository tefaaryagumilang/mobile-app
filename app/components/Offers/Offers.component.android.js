import React from 'react';
import PropTypes from 'prop-types';
import {Platform, View, ScrollView, PermissionsAndroid} from 'react-native';
import styles from './Offers.styles';
import result from 'lodash/result';
import noop from 'lodash/noop';
import {language} from '../../config/language';
import {ScrollableTabBar, ScrollableTabView} from '@valdio/react-native-scrollable-tabview';
import TabOffers from '../QRPromo/tabOffers.component';
import TabMerchant from '../QRPromo/tabMerchant.component';
import {theme} from '../../styles/core.styles';
import {wrapObjectInFunction} from '../../utils/transformer.util';

let LocationServices;

if (Platform.OS === 'android') {
  LocationServices = require('react-native-android-location-services-dialog-box').default;
}

const tabBarConfig = {
  tabBarBackgroundColor: theme.white,
  tabBarActiveTextColor: theme.black,
  tabBarInactiveTextColor: theme.textGrey,
  tabBarUnderlineStyle: {
    backgroundColor: theme.brand
  },
  tabBarTextStyle: styles.tabText
};

class Offers extends React.Component {
  static propTypes = {
    offers: PropTypes.array.isRequired,
    offerID: PropTypes.string,
    onOfferClick: PropTypes.func,
    clickedOffer: PropTypes.object,
    closeHandler: PropTypes.func,
    isLogin: PropTypes.bool,
    changeTab: PropTypes.func,
    qrPromo: PropTypes.array,
    onPromoClick: PropTypes.func,
    merchantList: PropTypes.array,
    goToMaps: PropTypes.func,
    getMerchantList: PropTypes.func,
  }
  static defaultProps = {
    offers: [],
    onOfferClick: noop
  }

  state = {
    tabsRef: {}
  }

  renderTabBar = wrapObjectInFunction(<ScrollableTabBar />)

  changeTab = (res) => {
    const {getMerchantList = noop} = this.props;
    if (Platform.OS === 'android') {
      res.i === 1 && PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION).then((response) => {
        if (response === 'granted') {
          LocationServices.checkLocationServicesIsEnabled({
            message: language.QR_PROMO__ACCESS_LOCATION,
            ok: language.GENERIC__YES,
            cancel: language.GENERIC__NO,
            enableHighAccuracy: false,
            showDialog: true,
            openLocationServices: true
          }).then(() => {
            getMerchantList();
          }).catch(() => {
            this.state.tabsRef.goToPage(0);
          });
        } else {
          this.state.tabsRef.goToPage(0);
        }
      });
    } else {
      res.i === 1 && getMerchantList();
    }
  }

  componentWillMount () {
    this.setState({tabsRef: this.refs.Tabs});
  }

  render () {
    const offers = result(this.props, 'offers', []);
    const {clickedOffer = {}, closeHandler = noop, isLogin, qrPromo = [], onOfferClick = noop, onPromoClick = noop, merchantList = [], goToMaps = noop} = this.props;
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.scrollContainer}>
        <ScrollableTabView {...tabBarConfig} locked={true} renderTabBar={this.renderTabBar} onChangeTab={this.changeTab} ref={'Tabs'}>
          <View style={styles.OffersContainer} tabLabel={language.QR_TAB_PROMO}>
            <TabOffers  offers={offers} clickedOffer={clickedOffer}
              closeHandler={closeHandler} isLogin={isLogin} qrPromo={qrPromo} onOfferClick={onOfferClick} onPromoClick={onPromoClick}/>
          </View>
          <TabMerchant tabLabel={language.QR_TAB_MERCHANT} merchantList={merchantList} goToMaps={goToMaps}/>
        </ScrollableTabView>
      </ScrollView>
    );
  }
}

export default Offers;
