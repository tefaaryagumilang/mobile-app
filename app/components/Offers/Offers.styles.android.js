import {theme} from '../../styles/core.styles';
import {Dimensions} from 'react-native';

const {width} = Dimensions.get('window');
const trueWidth = width - 40;
const trueHeight = (trueWidth * 7) / 16;
const bannerHeight = (trueWidth * 6) / 10;

export default {
  container: {
    padding: 5,
    backgroundColor: theme.white
  },
  offerContainer: {
    borderWidth: 1,
    borderBottomLeftRadius: 8,
    borderBottomRightRadius: 8,
    borderColor: theme.disabledGrey,
    marginVertical: 10,
    backgroundColor: theme.contrast,

    elevation: 2,
    shadowOffset: {width: 5, height: 5},
    shadowColor: 'grey',
    shadowOpacity: 0.5,
    shadowRadius: 5,
  },
  offerDetails: {
    padding: 10,
    flexDirection: 'row',
    borderTopWidth: 0.5,
    borderTopColor: theme.greyLine
  },
  offerImage: {
    width: trueWidth,
    height: trueHeight
  },
  separator: {
    flex: 1,
    height: 1,
    backgroundColor: theme.brand,
  },
  separatorContainer: {
    flexDirection: 'row',
    paddingVertical: 5,
    alignItems: 'center',
  },
  separatorText: {
    textAlign: 'center',
    flex: 1,
    color: theme.brand,
    fontWeight: theme.fontWeightMedium,
    fontSize: theme.fontSizeNormal
  },
  crossButtonWrapper: {
    padding: 20,
  },
  iconStyle: {
    color: 'white'
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: theme.brand
  },
  noOffers: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  headerText: {
    color: 'white',
    padding: 20,
    fontSize: theme.fontSizeMedium
  },
  scrollContainer: {
    paddingBottom: 80
  },
  promoDetailContainer: {
    padding: 10,
    flexDirection: 'row'
  },
  promoAmountContainer: {
    backgroundColor: theme.brand,
    padding: 5,
    justifyContent: 'center',
  },
  promoDetailDateContainer: {
    flex: 4,
    justifyContent: 'center',
    paddingHorizontal: 10
  },
  discountType: {
    fontSize: theme.fontSizeSmall,
    color: theme.white,
    fontWeight: theme.fontWeightMedium
  },
  discountAmount: {
    color: theme.white,
    fontWeight: theme.fontWeightBold
  },
  boldText: {
    fontWeight: theme.fontWeightBold
  },
  tabbar: {
    backgroundColor: theme.white,
  },
  indicator: {
    backgroundColor: theme.brand,
  },
  tabTextActive: {
    color: theme.black
  },
  qrTag: {
    justifyContent: 'center',
  },
  qrTagText: {
    color: theme.qrPromoTurqoise,
    fontWeight: theme.fontWeightBold,
    fontStyle: 'italic'
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingTop: 10,
    paddingHorizontal: 20
  },
  containerNoPadding: {
    paddingVertical: 10,
    backgroundColor: theme.containerGrey,
  },
  containerEgift: {
    backgroundColor: theme.containerGrey
  },
  OffersContainer: {
    paddingHorizontal: 14,
    paddingVertical: 10,
  },
  textstyle: {
    paddingHorizontal: 20,
    paddingTop: 10
  },
  seeAll: {
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
    width: 65,
    height: 25,
    borderRadius: 20,
    marginHorizontal: 10,
    position: 'absolute',
    right: 10,
    bottom: 45,
    alignSelf: 'flex-end',
  },
  burgerMenuContainer: {
    backgroundColor: 'rgba(252, 252, 245, 0.5)',
    width: 40,
    height: 40,
    borderRadius: 5,
    marginHorizontal: 18,
    marginVertical: 10,
    position: 'absolute',
    alignSelf: 'flex-start',
  },
  seeAlltext: {
    color: theme.white,
    alignSelf: 'center',
    paddingTop: 5
  },
  offerBannerContainer: {
    width: width,
    height: bannerHeight,
  },
  greyLine3: {
    height: 20,
    backgroundColor: theme.greyLine,
  },
  offersStylesImage: {
    borderWidth: 1,
    borderRadius: 30,
    borderColor: theme.disabledGrey,
  },
  offerImageBanner: {
    width: trueWidth,
    height: trueHeight,
  },
  bannerFlex: {
    flex: 1
  },
  label: {
    marginHorizontal: 10,
    fontSize: theme.fontSizeSmall,
    color: theme.black
  },
  labelValidDate: {
    marginHorizontal: 10,
    color: theme.black
  },
  iconStyleBlack: {
    color: theme.black
  },
  greyLine: {
    height: 10,
    backgroundColor: theme.greyLine,
    marginVertical: 15
  },
  greyLine2: {
    height: 10,
    backgroundColor: theme.red,
    marginVertical: 50,
  },
  burger: {
    color: theme.black,
    paddingHorizontal: 10,
    paddingVertical: 10
  },
  bannercontainer: {
    marginVertical: 6
  },
  categoryTabContainer: {
    flexDirection: 'row',
    marginHorizontal: 15,
    marginBottom: 5,
    marginTop: 25
  },
  categoryActive: {
    paddingHorizontal: 10,
    paddingVertical: 5,
    backgroundColor: theme.pinkBrand,
    borderRadius: 20,
    marginVertical: 5,
    marginHorizontal: 5,
    justifyContent: 'center'
  },
  categoryInActive: {
    paddingHorizontal: 10,
    paddingVertical: 5,
    backgroundColor: theme.white,
    borderRadius: 20,
    marginHorizontal: 5,
    marginVertical: 5,
    borderWidth: 1,
    borderColor: theme.grey,
    justifyContent: 'center'

  },
  activeText: {
    color: theme.white
  },
  inActiveText: {

  }
};
