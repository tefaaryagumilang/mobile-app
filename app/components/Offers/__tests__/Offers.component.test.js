import React from 'react';
import renderer from 'react-test-renderer';

import Offers from '../Offers.component';

describe('Offers component', () => {
  it('Offers: renders correctly', () => {
    const tree = renderer.create(<Offers />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
