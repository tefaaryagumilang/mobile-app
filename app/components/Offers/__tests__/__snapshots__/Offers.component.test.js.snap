// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers component Offers: renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "marginBottom": 80,
      "padding": 5,
    }
  }
>
  <View
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {},
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 1,
              "height": 50,
            },
            Object {
              "backgroundColor": "#FFFFFF",
            },
            Object {},
          ]
        }
      >
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          bounces={false}
          contentContainerStyle={
            Object {
              "flexGrow": 1,
              "justifyContent": "center",
            }
          }
          directionalLockEnabled={true}
          horizontal={true}
          scrollsToTop={false}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
        >
          <View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  },
                  Object {
                    "width": null,
                  },
                  Object {},
                ]
              }
            >
              <View
                accessibilityLabel="Promo"
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 49,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                      },
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontWeight": "bold",
                        },
                        undefined,
                      ]
                    }
                  >
                    Promo
                  </Text>
                </View>
              </View>
              <View
                accessibilityLabel="Merchant"
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 49,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                      },
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#9E9E9E",
                          "fontWeight": "normal",
                        },
                        undefined,
                      ]
                    }
                  >
                    Merchant
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#FA3F70",
                    "bottom": 0,
                    "height": 4,
                    "left": 0,
                    "position": "absolute",
                    "width": 0,
                  }
                }
              />
            </View>
          </View>
        </RCTScrollView>
      </View>
      <RCTScrollView
        alwaysBounceVertical={false}
        automaticallyAdjustContentInsets={false}
        contentOffset={
          Object {
            "x": 0,
          }
        }
        directionalLockEnabled={true}
        horizontal={true}
        keyboardDismissMode="on-drag"
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        pagingEnabled={true}
        scrollEnabled={false}
        scrollEventThrottle={16}
        scrollsToTop={false}
      >
        <View>
          <View
            style={
              Object {
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
              tabLabel="Promo"
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text>
                    No Offers for this Category
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "width": 750,
              }
            }
          >
            <View
              tabLabel="Merchant"
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
